MovieTime/MovieTime/Core/MTNetworkError.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  MTNetworkError.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 08/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import Foundation
+
+final class MTNetworkError: Error {
+    var statusCode: Int
+    var message: String
+
+    init(code: Int = -1, message: String? = nil) {
+        self.statusCode = code
+        self.message = message ?? "An unknown error happened."
+    }
+
+    static func unknownError() -> MTNetworkError {
+        return MTNetworkError()
+    }
+
+    static func parseError() -> MTNetworkError {
+        return MTNetworkError(code: -1001, message: "Internal server error.")
+    }
+
+    static func noConnection() -> MTNetworkError {
+        return MTNetworkError(code: -1001, message: "No internet connection.")
```

Comment:
why "internal server error" and "no internet connection" both have error code -1001?

Reply:
They should not be the same. It was a typing error. Every error should have a different code for better identification.

Reply:
in order to more easily avoid this, you could create an `enum ErrorType: Int` then each case must have a different rawValue, then you wouldn't use "codes" but `ErrorType`s instead